File: core/src/main/java/com/neqsoft/reso/cpu/CpuInfo.java
package com.neqsoft.reso.cpu;

import java.util.List;

import androidx.annotation.Nullable;

public class CpuInfo {

  @Nullable
  private String name;

  @Nullable
  private String architecture;

  private int coreCountPhysical;
  private int coreCountLogical;

  private int l1CacheSizeInBytes;
  private int l2CacheSizeInBytes;

  private Endianess endianess;

  private long frequency;

  @Nullable
  private List<Process> processes;
}

File: core/src/main/java/com/neqsoft/reso/cpu/CpuInfo.kt
package com.neqsoft.reso.cpu

data class CpuInfo(
        val name: String? = null,
        val architecture: String? = null,
        val coreCountPhysical: Int = -1,
        val coreCountLogical: Int = -1,
        val l1CacheSizeInBytes: Int = -1,
        val l2CacheSizeInBytes: Int = -1,
        val endianess: Endianess? = null,
        val frequency: Long = -1,
        val processes: List<Process>? = null
)